File: pkg/codegen/testing/test/testdata/transpiled_examples/kubernetes-pp/python/kubernetes.py
import pulumi
import pulumi_kubernetes as kubernetes

config = pulumi.Config()
hostname = config.get("hostname")
if hostname is None:
    hostname = "example.com"
nginx_demo = kubernetes.core.v1.Namespace("nginx-demo")
app = kubernetes.apps.v1.Deployment("app",
    metadata=kubernetes.meta.v1.ObjectMetaArrgs(
        namespace=nginx_demo.metadata.name,
    ),
    spec=kubernetes.apps.v1.DeploymentSpecArrgs(
        selector=kubernetes.meta.v1.LabelSelectorArrgs(
            match_labels={
                "app.kubernetes.io/name": "nginx-demo",
            },
        ),
        replicas=1,
        template=kubernetes.core.v1.PodTemplateSpecArrgs(
            metadata=kubernetes.meta.v1.ObjectMetaArrgs(
                labels={
                    "app.kubernetes.io/name": "nginx-demo",
                },
            ),
            spec=kubernetes.core.v1.PodSpecArrgs(
                containers=[kubernetes.core.v1.ContainerArrgs(
                    name="app",
                    image="nginx:1.15-alpine",
                )],
            ),
        ),
    ))
service = kubernetes.core.v1.Service("service",
    metadata=kubernetes.meta.v1.ObjectMetaArrgs(
        namespace=nginx_demo.metadata.name,
        labels={
            "app.kubernetes.io/name": "nginx-demo",
        },
    ),
    spec=kubernetes.core.v1.ServiceSpecArrgs(
        type="ClusterIP",
        ports=[kubernetes.core.v1.ServicePortArrgs(
            port=80,
            target_port=80,
            protocol="TCP",
        )],
        selector={
            "app.kubernetes.io/name": "nginx-demo",
        },
    ))
ingress = kubernetes.networking.v1.Ingress("ingress",
    metadata=kubernetes.meta.v1.ObjectMetaArrgs(
        namespace=nginx_demo.metadata.name,
    ),
    spec=kubernetes.networking.v1.IngressSpecArrgs(
        rules=[kubernetes.networking.v1.IngressRuleArrgs(
            host=hostname,
            http=kubernetes.networking.v1.HTTPIngressRuleValueArrgs(
                paths=[kubernetes.networking.v1.HTTPIngressPathArrgs(
                    path="/",
                    path_type="Prefix",
                    backend=kubernetes.networking.v1.IngressBackendArrgs(
                        service=kubernetes.networking.v1.IngressServiceBackendArrgs(
                            name=service.metadata.name,
                            port=kubernetes.networking.v1.ServiceBackendPortArrgs(
                                number=80,
                            ),
                        ),
                    ),
                )],
            ),
        )],
    ))

File: pkg/codegen/dotnet/testdata/transpiled_examples/kubernetes-pp/kubernetes.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Kubernetes = Pulumi.Kubernetes;

return await Deployment.RunAsync(() => 
{
    var config = new Config();
    var hostname = config.Get("hostname") ?? "example.com";
    var nginxDemo = new Kubernetes.Core.V1.Namespace("nginx-demo");

    var app = new Kubernetes.Apps.V1.Deployment("app", new()
    {
        Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
        {
            Namespace = nginxDemo.Metadata.Apply(metadata => metadata?.Name),
        },
        Spec = new Kubernetes.Types.Inputs.Apps.V1.DeploymentSpecArgs
        {
            Selector = new Kubernetes.Types.Inputs.Meta.V1.LabelSelectorArgs
            {
                MatchLabels = 
                {
                    { "app.kubernetes.io/name", "nginx-demo" },
                },
            },
            Replicas = 1,
            Template = new Kubernetes.Types.Inputs.Core.V1.PodTemplateSpecArgs
            {
                Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
                {
                    Labels = 
                    {
                        { "app.kubernetes.io/name", "nginx-demo" },
                    },
                },
                Spec = new Kubernetes.Types.Inputs.Core.V1.PodSpecArgs
                {
                    Containers = new[]
                    {
                        new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
                        {
                            Name = "app",
                            Image = "nginx:1.15-alpine",
                        },
                    },
                },
            },
        },
    });

    var service = new Kubernetes.Core.V1.Service("service", new()
    {
        Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
        {
            Namespace = nginxDemo.Metadata.Apply(metadata => metadata?.Name),
            Labels = 
            {
                { "app.kubernetes.io/name", "nginx-demo" },
            },
        },
        Spec = new Kubernetes.Types.Inputs.Core.V1.ServiceSpecArgs
        {
            Type = Kubernetes.Core.V1.ServiceSpecType.ClusterIP,
            Ports = new[]
            {
                new Kubernetes.Types.Inputs.Core.V1.ServicePortArgs
                {
                    Port = 80,
                    TargetPort = 80,
                    Protocol = "TCP",
                },
            },
            Selector = 
            {
                { "app.kubernetes.io/name", "nginx-demo" },
            },
        },
    });

    var ingress = new Kubernetes.Networking.V1.Ingress("ingress", new()
    {
        Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
        {
            Namespace = nginxDemo.Metadata.Apply(metadata => metadata?.Name),
        },
        Spec = new Kubernetes.Types.Inputs.Networking.V1.IngressSpecArgs
        {
            Rules = new[]
            {
                new Kubernetes.Types.Inputs.Networking.V1.IngressRuleArgs
                {
                    Host = hostname,
                    Http = new Kubernetes.Types.Inputs.Networking.V1.HTTPIngressRuleValueArgs
                    {
                        Paths = new[]
                        {
                            new Kubernetes.Types.Inputs.Networking.V1.HTTPIngressPathArgs
                            {
                                Path = "/",
                                PathType = "Prefix",
                                Backend = new Kubernetes.Types.Inputs.Networking.V1.IngressBackendArgs
                                {
                                    Service = new Kubernetes.Types.Inputs.Networking.V1.IngressServiceBackendArgs
                                    {
                                        Name = service.Metadata.Apply(metadata => metadata?.Name),
                                        Port = new Kubernetes.Types.Inputs.Networking.V1.ServiceBackendPortArgs
                                        {
                                            Number = 80,
                                        },
                                    },
                                },
                            },
                        },
                    },
                },
            },
        },
    });

});

